src/ml_gym/util/util.py

```diff
@@ -234,25 +255,45 @@ def update_evaluation_details(exp_config: dict) -> EvalDetails:
 
                 return EvalDetails(loss_funcs = loss_funcs, metrics = metrics)
             except Exception as e:
-                print("Error while fetching Eval Details for Model card.", e)
-                return EvalDetails()
+                raise EvalDetailsCreationError(f"Error while fetching Eval Details for Model card.") from e
+        
+        def update_pipeline_details(exp_config: dict) -> PipelineDetails:
+            """
+            Function to initialize PipelineDetails object.
+            :params:
+                    exp_config (dict): Experiment configuration.
+            :returns:
+                    obj (PipelineDetails): initialized pipeline details object.
+            """
+            try:
+                pipeline_details = {}
+                found_keys = set()
+                all_keys = set(key for key in exp_config)
+                find_key_arr = ["data", "model", "eval", "train", "early_stopping", "checkpointing"]
```

Comment:
What do you think about going through the gs_config recursively (it's a tree structure) and just take the "leaf" components for the pipeline details. In that case, we don't have the find_key_arr hardcoded here. 
What do you think?

Reply:
Yes, it is possible but not with the current yaml structure. Ex: now you have nodes in the tree like dataset_repository, dataset_iterators, splitted_dataset_iterators, data_collator, data_loaders. All of these nodes are connected to the root, if I want to summarize the information like data, model, eval, ... unfortunately, this level is not present in the tree. So I am getting the JSON and beautifying it for ease of view and access.

if I wanted to only take the leaf in the yaml then I would need to follow yaml structure:

data:
    dataset_iterators:
     .....
     splitted_dataset_iterators:
     .....
model:
     model_registry:
      .....

something like this. To a layman, the config is not readable and needs a lot of documentation to understand and use in the project so summarizing and keeping separation between different tasks in the pipeline is important according to me.

Reply:
yeah showing them at the same time, might be not a good option. But we could go with collapsible elements in the UI. So given the length of the discussion, I think we need a decision here. I'm strongly in favor of forwarding all components (without aggregation) to the client UI and go with collapsible elements there , that reduce the complexity. If you don't have pressing objections, then let's go with that. 

FYI,  @vijulshah

Reply:
I have removed all the aggregation, I am sending the entire experiment config as a JSON response in the pipeline_details key. If there is any aggregation required it can be performed in the UI. I have also commented the aggregation code in the backend but did not remove it yet, so it will be available if needed.